apps/passport-client/components/screens/MultiChoiceScanScreen.tsx

```diff
@@ -0,0 +1,219 @@
+
+  button {
+    flex-grow: 1;
+  }
+`;
+
+// Scan a PCD QR code, then go to /verify to verify and display the proof.
+export function MultiChoiceScanScreen(): JSX.Element {
+  const usingLaserScanner = loadUsingLaserScanner();
+  useLaserScannerKeystrokeInput();
+  const nav = useNavigate();
+  const dispatch = useDispatch();
+
+  const [scanner, setScanner] = useState<
+    "strich" | "react-qr-reader" | "scandit"
+    // @ts-expect-error this is temporary code so doesn't need to type-check
+  >(localStorage.getItem("preferred-scanner") ?? "scandit");
+
+  type Option = {
+    id: "strich" | "react-qr-reader" | "scandit";
+    label: string;
+  };
+
+  const onChange = useCallback(
+    (option: SingleValue<Option>) => {
+      if (option) {
+        setScanner(option.id);
+        localStorage.setItem("preferred-scanner", option.id);
+      }
+    },
+    [setScanner]
+  );
+
+  const options: Option[] = [
+    { id: "scandit", label: "Scandit" },
+    { id: "strich", label: "Strich" },
+    { id: "react-qr-reader", label: "React-QR-Reader" }
+  ];
+
+  const onResult = useCallback(
+    (result: string): void => {
+      console.log(`Got result, considering redirect`, result);
+      const newLoc = maybeRedirect(result);
+      if (newLoc) {
+        nav(newLoc);
+      } else {
+        dispatch({
+          type: "error",
+          error: {
+            title: "Not a Zupass QR code",
+            message:
+              "The QR code you scanned is not a Zupass QR code. Make sure the QR code you're scanning comes from the Zupass app.",
+            dismissToCurrentPage: true
+          }
+        });
+      }
+    },
+    [dispatch, nav]
+  );
+
+  const onQrReaderResult = useCallback(
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    // @ts-ignore
+    (result) => {
+      if (result) {
+        onResult(result.getText());
+      }
+    },
+    [onResult]
+  );
+
+  return (
+    <AppContainer bg="gray">
+      {!usingLaserScanner && (
+        <QRContainer>
+          <Spacer h={8} />
+          <ButtonsContainer>
+            <Back />
+            <Home />
+          </ButtonsContainer>
+          <Select
+            value={options.find((option) => option.id === scanner)}
+            onChange={onChange}
+            options={options}
+          ></Select>
+          <Spacer h={16} />
+          {scanner === "react-qr-reader" && (
+            <QrReader
+              className="qr"
+              onResult={onQrReaderResult}
+              constraints={{ facingMode: "environment", aspectRatio: 1 }}
+              ViewFinder={ViewFinder}
+              containerStyle={{ width: "100%" }}
+            />
+          )}
+          {scanner === "scandit" && <ScanditScanner onScan={onResult} />}
+          {scanner === "strich" && <Scanner onResult={onResult} />}
+          <Spacer h={16} />
+          <TextCenter>Scan a ticket</TextCenter>
+        </QRContainer>
+      )}
+      {usingLaserScanner && (
```

Comment:
relevant for @rrrliu https://linear.app/0xparc-pcd/issue/0XP-505/ensure-sunmi-laser-scanning-flow-still-works